osquery/database/database.cpp

```diff
@@ -493,7 +493,7 @@ Status DatabasePlugin::call(const PluginRequest& request,
   auto key = (request.count("key") > 0) ? request.at("key") : "";
 
   if (request.at("action") == "reset") {
-    WriteLock lock(kDatabaseReset);
+    // resetDatabase is expected to hold write lock
```

Comment:
I'm not sure, the crashes you're observing are from a regression from https://github.com/facebook/osquery/commit/e52f85f95e34b656d0f25e6a94350b03f75edafb

But that fixes a rare crash when an extension uses the Registry/Plugin APIs to call `reset` outside of the `resetDatabase` method.